File: SPOJ/java/COMDIV.java
import java.io.*;
import java.util.*;
import static java.lang.Math.min;
import static java.lang.Math.sqrt;
import static java.lang.Integer.parseInt;

class Main{

	public static void main(String[] args) throws IOException{	
		int N = 3000;
		boolean[] primes = new boolean[N+1];
		for(int i=2;i<=N;i++)
			primes[i] = true;
		for(int i=2;i<=sqrt(N);i++)
			if(primes[i])
				for(int j=i*i;j<=N;j+=i)
					primes[j]=false;


		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int T = parseInt(br.readLine());
		for(int t=0;t<T;t++)
		{
			StringTokenizer st = new StringTokenizer(br.readLine());
			int A = parseInt(st.nextToken());	
			int B = parseInt(st.nextToken());	
			int ret = 1;
			for(int i=2;i<=N;i++)
				if(primes[i])
				{
					int a = A; int ac = 0;
					while(a%i==0){a=a/i;ac++;}
					int b = B; int bc = 0;
					while(b%i==0){b=b/i;bc++;}
					ret = ret * (min(ac,bc)+1);
				}
			System.out.println(ret);
		}
		br.close();
	}

}

File: SPOJ/cpp/COMDIV.c
#include<stdio.h>
#define N 1021
#define min(A,B) (((A)<(B))?(A):(B))
int P[N];
int PP[N];
int PSIZE = 0;

void primes(){
	int i,j;
	for(i=2;i<=N;i++)
		P[i]=1;
	for(i=2;i*i<=N;i++)
		if(P[i])
			for(j=i*i;j<=N;j+=i)
				P[j]=0;
	for(i=2;i<=N;i++)
		if(P[i]) PP[PSIZE++] = i;
	for(i=0;i<PSIZE;i++)
		printf("%d,",PP[i]);
	printf("\n");

}

int gcd(int a, int b){
	return (b==0)?a:gcd(b,a%b);
}

int main(){
	int T,A,B,i,a,ai,b,bi,ret,p,g;
	primes();
	scanf("%d",&T);
	while(T--){
		scanf("%d%d",&A,&B);
		ret = 1;
		g = gcd(A,B);
		for(i=0;i<PSIZE&&PP[i]<=g;i++)
			{
				p = PP[i];
				a=g;ai=0;
				while((a%p)==0){ai++;a/=p;}
				ret *= (ai+1);
			}
		printf("%d\n",ret);
	}
}

